### PR TITLE
feat: error handling feature

### LINKED ISSUE #219: Add error handing for scripts
### Describe the feature request

The scripts could be made more robust by implementing better error handling. This would involve checking the return values of commands and taking appropriate actions, such as printing helpful error messages and exiting gracefully.

If the team would welcome this, I'd be happy to work on it.

### Spam policy

- [X] I verify that this issue is NOT SPAM and understand SPAM issues will be closed and reported to GitHub, resulting in ACCOUNT TERMINATION.

### PR DESCRIPTION
**Pull request by richardgreg**
Add error-handling and retry logic to scripts

Closes #219

### CODE REVIEW

File: script/start-prover-relayer.sh [hunk 1]
```diff
@@ -87,8 +92,10 @@ if [ "$ENABLE_PROVER" = "true" ]; then
         ARGS="${ARGS} --tx.sendTimeout ${TX_SEND_TIMEOUT}"
     fi
 
-    exec taiko-client prover ${ARGS}
-else
-    echo "PROVER IS DISABLED"
-    sleep infinity
-fi
+    # Run the prover with error handling
+    if [ "$ENABLE_PROVER" = "true" ]; then
```
Comment (RogerLamTd) [inline script/start-prover-relayer.sh:96]: @richardgreg this condition is already checked here in [L11](https://github.com/taikoxyz/simple-taiko-node/blob/11e77ea27878aa388c173bbe1b06d434c4cc5180/script/start-prover-relayer.sh#L11), also i believe you're missing an `fi` at the end of the script

Comment (RogerLamTd) [reply script/start-prover-relayer.sh:96]: Also, could refactor the handle_error() function into a util script and just `source` it into the start-prover-relayer and start-proposer scripts :)

File: script/start-proposer.sh [hunk 2]
```diff
@@ -78,9 +80,17 @@ if [ -n "$TX_SEND_TIMEOUT" ]; then
...
-    echo "PROPOSER IS DISABLED"
-    sleep infinity
-fi
+# Retry logic parameters
+MAX_RETRIES=3
+RETRY_DELAY=5
+
+# Run the proposer with retries and handle error errors
+for i in $(seq 1 $MAX_RETRIES); do
+  taiko-client proposer ${ARGS} && break  # Exit loop on success
```
Comment (RogerLamTd) [inline script/start-proposer.sh:89]:
```suggestion
  exec taiko-client proposer ${ARGS} && break  # Exit loop on success
```

File: script/start-prover-relayer.sh [hunk 3]
```diff
@@ -87,7 +88,8 @@ if [ "$ENABLE_PROVER" = "true" ]; then
         ARGS="${ARGS} --tx.sendTimeout ${TX_SEND_TIMEOUT}"
     fi
 
-    taiko-client prover ${ARGS}
+    # Run the prover with error handling
+    taiko-client prover ${ARGS} || handle_error "Failed to start prover. Check the logs for details."
```
Comment (RogerLamTd) [inline script/start-prover-relayer.sh:92]:
```suggestion
    exec taiko-client prover ${ARGS} || handle_error "Failed to start prover. Check the logs for details."
```

File: script/start-proposer.sh (hunk 2 above)
Comment (RogerLamTd) [reply script/start-proposer.sh:89]: ref #224

**Comment by RogerLamTd** [comment]
@richardgreg lgtm, please resolve conflicts and should be good to merge!

**Comment by RogerLamTd** [comment]
> Is it safe to assume this PR is no longer relevant? 😁

Sorry, been busy with other stuff; could you fix the conflicts and test the error handling locally perhaps? Just ensure that it functions as intended with docker :)

**Review by flechemano — APPROVED** [review]

**Comment by RogerLamTd** [comment]
closing as stale, please feel free to reopen with changes/tests.

**Pull request closed without merging**